s3/pkg/datastore/common/utils.go

```diff
@@ -13,6 +13,14 @@ func GetMd5FromCtx(ctx context.Context) (md5val string) {
 	return
 }
 
+func GetStorClassFromCtx(ctx context.Context) (storClass string) {
+	// StorageClass Provided by user for object PUT
+	if val := ctx.Value(CONTEXT_KEY_STORAGE_CLASS); val != nil {
+		return val.(string)
```

Comment:
In my opinion. this might crash in error scenario like if Storage class value passed other than string

Reply:
right. The Storage Class recieved in X-Amz-Storage-Class will always be sent as string from the API. 
this will be part of swagger validation too, anyways
To double check, have tested with this value being sent as string, int, float..
If the user sets wrong value, the s3 service layer will handle it